src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,142 @@
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.ofNullable(extractBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not fetch the book with the provided ID: " + id);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql);
+        ) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> books = new ArrayList<>();
+            while (resultSet.next()) {
+                books.add(extractBook(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not fetch books");
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql);
+        ) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException(
+                        "Expected to update two rows, but updated 0 rows."
+                );
+            } else {
+                return book;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not update the book: " + book);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        Optional<Book> deletingBook = findById(id);
+        if (deletingBook.equals(Optional.empty())) {
+            throw new DataProcessingException(
+                    "Impossible to delete the book with provided ID: "
+                            + id
+                            + ", because there isn't this book in the DB"
+            );
+        }
+
+        String sql = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql);
+        ) {
+            statement.setLong(1, id);
+            int affectedRows = statement.executeUpdate();
+            return affectedRows > 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can not delete the book with the provided ID: " + id
+            );
+        }
+    }
+
+    private Book extractBook(ResultSet resultSet) {
+        try {
+            if (resultSet.next()) {
```

Comment:
```suggestion
            String title = resultSet.getString("title");
            BigDecimal price = resultSet.getBigDecimal("price");
            Long id = (long) resultSet.getObject("id");
            return new Book(id, title, price);
```

Reply:
You call this method inside `while (resultSet.next())` so don't need `if (resultSet.next()) {` here